src/imgtools/io/sample_input.py

```diff
@@ -0,0 +1,533 @@
+                )
+                logger.warning(msg, folder=folder, modality=modality)
+            file_name_sets.append(
+                [
+                    (root_dir / file_name).as_posix()
+                    for subseries in series_info
+                    for file_name in series_info[subseries][
+                        "instances"
+                    ].values()
+                ]
+            )
+
+        # load the series
+        return [
+            read_dicom_auto(
+                path=root_dir.as_posix(),
+                modality=modality,
+                file_names=file_name_set,
+                series_id=series_uid,
+            )
+            for file_name_set in file_name_sets
+        ]
+
+    def load_sample(  # noqa: PLR0912
+        self,
+        sample: Sequence[SeriesNode],
+        load_subseries: bool = False,
+    ) -> Sequence[MedImage | VectorMask]:
+        # group list by modality
+        by_mod: defaultdict[str, list[SeriesNode]] = defaultdict(list)
+        for series in sample:
+            by_mod[series.Modality].append(series)
+
+        reference_modality = (
+            "CT"
+            if "CT" in by_mod
+            else "MR"
+            if "MR" in by_mod
+            else "PT"
+            if "PT" in by_mod
+            else None
+        )
+        if not reference_modality:
+            raise ValueError(
+                "No CT, MR, or PT series found to use as reference."
+            )
+        if len(by_mod[reference_modality]) > 1:
+            msg = (
+                f"Found {len(by_mod[reference_modality])} "
+                f"{reference_modality} series, "
+                "using the first one as reference."
+            )
+            logger.warning(msg, reference_list=by_mod[reference_modality])
+
+        # Load the first found reference series
+        reference_series = by_mod[reference_modality].pop(0)
+        reference_images = self._read_series(
+            series_uid=reference_series.SeriesInstanceUID,
+            modality=reference_modality,
+            folder=reference_series.folder,
+            load_subseries=load_subseries,
+        )
+
+        images: Sequence[MedImage] = []
+
+        # hack to satisfy mypy for now
+        reference_scans = cast("list[MedImage]", reference_images)
+
+        # Extract the first (of possibly many subseries) as the reference image
+        reference_image = reference_scans[0]
+        images = reference_scans[1:]
+        # Load the rest of the series
+        for modality, series_nodes in by_mod.items():
+            if modality == reference_modality:
+                # TODO:: maybe implement some check here in case we loading
+                # another of same reference modality?
+                continue
+
+            # for all other series, we load them and insert them into the
+            # images list
+            self._load_non_ref_images(
```

Comment:
Unnecessary side effect, better to do:
```suggestion
            image = self._load_non_ref_image()
            images.append(image)
```